fpm/src/fpm_compiler.f90

```diff
@@ -0,0 +1,198 @@
+
+type(fpm_model_t), intent(inout) :: model
+! could just be a function to return a string instead of passing model 
+! but likely to change other components like matching C compiler
+     
+character(len=:),allocatable :: fflags
+character(len=:),allocatable :: module_path_switch
+
+! special reserved names "debug" and "release" are for supported compilers with no user-specified compile or load flags
+
+! vendor            Fortran   C         Module output   Module include OpenMP    Free for OSS
+!                   compiler  compiler  directory       directory
+! Gnu               gfortran   gcc     -J              -I            -fopenmp   X
+! Intel             ifort      icc     -module         -I            -qopenmp   X
+! Intel(Windows)    ifort      icc     /module:path    /I            /Qopenmp   X
+! Intel oneAPI      ifx        icx     -module         -I            -qopenmp   X
+! PGI               pgfortran  pgcc    -module         -I            -mp        X
+! NVIDIA            nvfortran  nvc     -module         -I            -mp        X
+! LLVM flang        flang      clang   -module         -I            -mp        X
+! LFortran          lfortran   ---     ?               ?             ?          X
+! Lahey/Futjitsu    lfc        ?       -M              -I            -openmp    ?
+! NAG               nagfor     ?       -mdir           -I            -openmp    x
+! Cray              crayftn    craycc  -J              -I            -homp      ?
+! IBM               xlf90      ?       -qmoddir        -I            -qsmp      X
+! Oracle/Sun        ?          ?       -moddir=        -M            -xopenmp   ?
+! Silverfrost FTN95 ftn95      ?       ?               /MOD_PATH     ?          ?
+! Elbrus            ?          lcc     -J              -I            -fopenmp   ?
+! Hewlett Packard   ?          ?       ?               ?             ?          discontinued
+! Watcom            ?          ?       ?               ?             ?          discontinued
+! PathScale         ?          ?       -module         -I            -mp        discontinued
+! G95               ?          ?       -fmod=          -I            -fopenmp   discontinued
+! Open64            ?          ?       -module         -I            -mp        discontinued
+! Unisys            ?          ?       ?               ?             ?          discontinued
+
+    select case(build_name//'_'//compiler)
+
+    case('release_gfortran') 
+       module_path_switch='-J '
+       fflags='&
+       & -O3&
+       & -Wimplicit-interface&
+       & -fPIC&
+       & -fmax-errors=1&
+       & -ffast-math&
+       & -funroll-loops&
+       &'
+    case('debug_gfortran')
+       module_path_switch='-J '
+       fflags = '&
+       & -Wall&
+       & -Wextra&
+       & -Wimplicit-interface&
+       & -fPIC -fmax-errors=1&
+       & -g&
+       & -fbounds-check&
+       & -fcheck-array-temporaries&
+       & -fbacktrace&
+       &'
+
```

Comment:
The bootstrap version supports the OpenCoarrays `caf` compiler. I believe it has the same options as `gfortran` except without `-fcoarray=single`

<https://github.com/fortran-lang/fpm/blob/7d320295c6fe56cc767b352bd26f0466aa29ac5c/bootstrap/src/Fpm.hs#L662-L686>

Reply:
Just to confirm, `caf` is a thin wrapper around `gfortran` that simply adds the necessary compile and link flags. Which is why it has exactly the same flags.

Reply:
so I just add a case for "caf" in fpm_compile.f90, right?

```fortran
case('release_caf')
       module_path_switch='-J '
       fflags='&
       & -O3&
       & -Wimplicit-interface&
       & -fPIC&
       & -fmax-errors=1&
       & -ffast-math&
       & -funroll-loops&
       &'
    case('debug_caf')
       module_path_switch='-J '
       fflags = '&
       & -Wall&
       & -Wextra&
       & -Wimplicit-interface&
       & -fPIC -fmax-errors=1&
       & -g&
       & -fbounds-check&
       & -fcheck-array-temporaries&
       & -fbacktrace&
       &'
```